src/Storage/Collection/Relations.php

```diff
@@ -234,13 +236,39 @@ public function offsetGet($offset)
         return $collection;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function serialize()
     {
-        $output = [];
-        foreach ($this as $k => $existing) {
-            $output[$existing->getToContenttype()][] = spl_object_hash($existing);
+        $trace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 1);
+        if (isset($trace[0]['file'])) {
+            Deprecated::method(3.4, 'toArray');
+
+            $output = [];
+            foreach ($this as $k => $existing) {
+                $output[$existing->getToContenttype()][] = spl_object_hash($existing);
+            }
+
+            return $output;
```

Comment:
This logic is different than what `toArray` would return. We should not suggest it or move this logic to another method someone can call.